src/workerd/jsg/resource.h

```diff
@@ -44,6 +45,28 @@ const auto kIllegalInvocation =
 class Serializer;
 class Deserializer;
 
+// Helper function for executing fast methods with consistent error handling.
+// Used by FastMethodCallback to propagate errors in a consistent manner.
+template <typename Ret, typename Func>
+Ret executeFastMethod(v8::Isolate* isolate, Func&& func) {
+  try {
+    return func();
+  } catch (kj::Exception& exception) {
+    throwInternalError(isolate, kj::mv(exception));
+  } catch (JsExceptionThrown&) {
+    // Exception already set
+  } catch (std::exception& exception) {
+    throwInternalError(isolate, exception.what());
+  } catch (...) {
+    throwInternalError(
+        isolate, kj::str("caught unknown exception of type: ", kj::getCaughtExceptionType()));
+  }
+
+  if constexpr (!std::is_same_v<Ret, void>) {
```

Comment:
Prefer `kj::isSameType` for consistency -- lots of stuff in JSG and the rest of our codebase uses `kj::isSameType`, it's confusing if we add a bunch of other code using `std::is_same_v` (which does the same thing).